src/upgrade/upgrade_utils.js

```diff
@@ -35,39 +35,39 @@ function version_compare(ver1, ver2) {
 }
 
 /**
- * @param {string} server_version
- * @param {string} container_version
+ * @param {string} current_version
+ * @param {string} new_version
  */
-function should_upgrade(server_version, container_version) {
-    if (!server_version) {
+function should_upgrade(current_version, new_version) {
+    if (!current_version) {
         dbg.log('system does not exist. no need for an upgrade');
         return false;
     }
-    const ver_comparison = version_compare(container_version, server_version);
+    const ver_comparison = version_compare(new_version, current_version);
     if (ver_comparison === 0) {
-        if (server_version !== container_version) {
-            dbg.warn(`the container and server appear to be the same version but different builds. (container: ${container_version}), (server: ${server_version})`);
+        if (current_version !== new_version) {
+            dbg.warn(`the new_version and current_version (server) appear to be the same version but different builds. (new_version: ${new_version}), (current_version: ${current_version})`);
             dbg.warn(`upgrade is not supported for different builds of the same version!!`);
         }
-        dbg.log0('the versions of the container and the server match. no need to upgrade');
+        dbg.log0('the versions of the new_version and the current_version match. no need to upgrade');
         return false;
     } else if (ver_comparison < 0) {
-        // container version is older than the server version - can't downgrade
-        dbg.error(`the container version (${container_version}) appear to be older than the current server version (${server_version}). cannot downgrade`);
-        throw new Error('attempt to run old container version with newer server version');
+        // new_version is older than the current server version - can't downgrade
+        dbg.error(`the new_version (${new_version}) appears to be older than the current server version (${current_version}). cannot downgrade`);
```

Comment:
Minor (to be aligned with previous printings), if you don't want to change it is also fine.
```suggestion
        dbg.error(`the new_version (${new_version}) appears to be older than the current_version (server) (${current_version}). cannot downgrade`);
```